### PR TITLE
feat: collection buy event resolver

### PR DESCRIPTION
**Pull request by Jarsen136**
### Add a resolver to return `max` and `count` of collection's events.
More details ref: https://github.com/kodadot/nft-gallery/pull/2556

It's my first time contributing to the subsquid project. I encountered some problems connecting the local server, so I wrote the code directly and haven't tested it.

Please don't hesitate to point out my mistakes and give me some corrections. Thanks a lot! @vikiival

### CODE REVIEW

File: src/server-extension/query/event.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export const maxBuyEvent = `SELECT
+    date(e.timestamp),
+    COALESCE(MAX(e.meta::bigint), 0) as max,
```
Comment (vikiival) [inline src/server-extension/query/event.ts:3]:
```suggestion
    date(e.timestamp),
    COUNT(e.*) as count
    COALESCE(MAX(e.meta::bigint), 0) as max,
```

File: src/server-extension/query/event.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    FROM event e
+    LEFT JOIN nft_entity ne on ne.id = e.nft_id
+    WHERE e.interaction = 'BUY' AND ne.collection_id = $1;`
+
+export const totalBuyEvent = `SELECT
+    COUNT(e.*) as count
+    FROM event e
+    LEFT JOIN nft_entity ne on ne.id = e.nft_id
+    WHERE e.interaction = 'BUY' AND ne.collection_id = $1;`
```
Comment (vikiival) [inline src/server-extension/query/event.ts:12]:
```suggestion

```

**Review by vikiival — CHANGES_REQUESTED** [review]
Looks oki I would merge it into one query :)

**Comment by Jarsen136** [comment]
@vikiival  Thanks for your comments. I have change the code.

**Comment by Jarsen136** [comment]
Thanks a lot for your help!  ❤️

**Pull request merged**